Revamp Inference Providers doc (#1652)
huggingface/hub-docs: docs/api-inference/security.md

# Security & Compliance

## Data Security/Privacy

> Hugging Face does not store any user data for training purposes. We do not parse or store the request body or response when routing requests through Hugging Face. Logs are kept for debugging purposes for up to 30 days, but no user data or tokens are stored.

Reviewer: ```suggestion
Hugging Face does not store any user data for training purposes. We do not store the request body or response when routing requests through Hugging Face. Logs are kept for debugging purposes for up to 30 days, but no user data or tokens are stored.
```

Reviewer: (we might need to parse a bit in the future cc @SBrandeis)

For more information on how your data is handled, please refer to the Data Security Policies of each provider.

> Inference Provider API uses TLS/SSL to encrypt data in transit.

Reviewer: ```suggestion
Inference Providers routing uses TLS/SSL to encrypt data in transit.
```

## Hub Security

> The Hugging Face Hub, which Inference Provider API is part, is SOC2 Type 2 certified. For more on Hub security: https://huggingface.co/docs/hub/security. External providers are responsible for their own security measures, so please refer to their respective security policies for more details.

Reviewer: ```suggestion
The Hugging Face Hub, which Inference Providers is a feature of, is SOC2 Type 2 certified. For more on Hub security: https://huggingface.co/docs/hub/security. External providers are responsible for their own security measures, so please refer to their respective security policies for more details.
```
